### ISSUE TITLE
Enhanced Context UX improvements

### ISSUE THREAD

**Issue by taylorsperry** [sourcegraph/jetbrains#982]
We need to make it easier for Enterprise users to: 
1. Know that they need to add remote repos to get the best results from Cody
2. Discover the way to add those remote repos (right now, that's the ➕ button at the bottom of the chat window)
3. Add remote repos to Cody's chat context

Also need to have event level telemetry on all of these interactions (opening the multi-repo popup, adding/removing repos, clicking links to docs, how _many_ repos a user has as context for a given chat, etc)

**[Figma designs are here](https://www.figma.com/file/m9Yvgv6W3aPRLfXnxgwnEz/JetBrains?type=design&node-id=751-13946&mode=design&t=iqkryj4mvBKcUqjc-0)**

**Comment by dominiccooney** [comment]
FWIW I'm working on the repo picking aspect of this (part 3 above, Add remote repos to Cody's chat context.)

Parts 1 & 2 (1. Know that they need to add remote repos to get the best results from Cody, 2. Discover the way to add those remote repos) need design input, see the comments in Figma linked above.

**Comment by dominiccooney** [comment]
Latest Loom from @danielmarquespt https://www.loom.com/share/c087165e4a0546a7abda6d1d320c3670

WIP tracking:

Chat context zippy:
- [X] Chat context has vertical pencil, clock, question mark icon stack instead of horizontal plus, etc.
- [x] Repos have code host icons (GitLab, etc.)
- [ ] Chat context has secondary text "Multi repo, $n Repos"
- [ ] "Remote Repo(s)" &rArr; "$n Repos on/$hostname"

Add repository dialog:
- [x] Add repository dialog is a JBPopup like VCS people search (`platform/vcs-log/impl/src/com/intellij/vcs/log/ui/filter/MultilinePopupBuilder.java`)
- [x] Repo adder is a multiline textbox with autocomplete
- [X] Repo adder autocomplete has a code host icon
- [ ] Text field document should not generate agent events: `WARN - #c.s.c.a.CodyAgentClient - Cody by Sourcegraph: █ ProtocolTextDocumentWithUri: correcting invariant violation mock:/Dummy.txt (this.uri) !== mock:///Dummy.txt (this.underlying.uri)`
- [X] Inserted content should not be doubled around dots, slashes

Questions for @danielmarquespt :

- In "N Repos on/sourcegraph.sourcegraph.com" should "on" be primary, like in your mock, or secondary? What's the thinking with putting the count first? Will "Repos: N/on sourcegraph.sourcegraph.com" be easier to skim because "Repos" is always in the leading position?
- I notice "Local Project" is missing. This matches VSCode. Want to confirm this is expected.

**Comment by dominiccooney** [comment]
@danielmarquespt 

- See two questions above.
- I need the SVG icons for other code hosts: GitHub, and "neither GitHub nor GitLab".

**Comment by dominiccooney** [comment]
I'm going to make some executive decisions here, @danielmarquespt I added some more question to Figma, let me know if any of this is wrong:

- Enhanced context is more than just remote repos for Enterprise users, so I'm adding the checkbox back to the top item in the tree view.
- The "Chat Context/9 Repos, 9 Repos/demo.sourcegraph.com" language seems redundant to me so I'm going to simplify the second one.

Note, the extension does a bunch of work to implicitly include a default repo. I'm unsure if that is (a) not working, (b) working but not displayed. I'm going to worry about that later. But flagging that we may need to change the logic for forcibly including a repo, or take away the "master" checkbox on the list of remote repos.

**Comment by danielmarquespt** [comment]
Other improvements as discussed via DM on Slack:

- [ ] Add [tooltip](https://jetbrains.design/intellij/controls/tooltip/) on hover of "Chat Context" for enterprise users with:
```
About chat context
Select up to 10 repositories from [customer_instance_url] to use as reference context in this chat.

[Context Docs](https://sourcegraph.com/docs/cody/core-concepts/context)
```
- [ ] Dismiss the repo selection window when clicking outside of it
- [ ] Add "Cody Context" entry as part of the [Search Everywhere](https://www.jetbrains.com/help/idea/searching-everywhere.html) 
- [ ] Add "Cody Context" as part of the Keymap list on JetBrains settings

--- 

Answering the Pending questions:

> Enhanced context is more than just remote repos for Enterprise users, so I'm adding the checkbox back to the top item in the tree view.

Fine

> In "N Repos on/sourcegraph.sourcegraph.com" should "on" be primary, like in your mock, or secondary? What's the thinking with putting the count first? Will "Repos: N/on sourcegraph.sourcegraph.com" be easier to skim because "Repos" is always in the leading position?"

I prefer the last version you applied:

```
Chat Context X repos on [instance]
    Repositories
         example/example
          example/example2
```
 
> I notice "Local Project" is missing. This matches VSCode. Want to confirm this is expected.

As far as I understand that is always part of the context, so it should be included.

**Comment by dominiccooney** [comment]
Basic functionality has landed; I have filed #1322 for follow-up polish.
